### PR TITLE
705 mixed patch errors

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Generate 417 error for mixed EDM houston patching when EDM patch succeeds and houston patch fails
- Send back details of the fields that were written
- Fix a couple of bugs found in testing too

---

### CODE REVIEW

File: app/modules/encounters/resources.py [hunk 1]
```diff
@@ -140,22 +140,33 @@ def patch(self, args, encounter):
...
-                db.session, default_error_message='Failed to update Encounter details.'
-            )
+            if not edm_args:
+                # regular houston-patching
+                context = api.commit_or_abort(
+                    db.session,
+                    default_error_message='Failed to update Encounter details.',
+                )
+            else:
+                # irregular houston-patching, where we need to report that EDM data was set if houston setting failed
```
Comment (naknomum) [inline app/modules/encounters/resources.py:162]: this seems like the right result given the situation.  :+1:

**Review by naknomum — APPROVED** [review]
looks good.

File: app/modules/sightings/resources.py [hunk 2]
```diff
@@ -472,20 +475,17 @@ def patch(self, args, sighting):
             enc_json_data = []
             try:
                 for arg in edm_args:
-                    if (
-                        arg.get('path', None) == '/encounters'
-                        and arg.get('op', None) == 'add'
+                    if arg.get('path', None) == '/encounters' and (
```
Comment (hwindsor) [inline app/modules/sightings/resources.py:478]: @naknomum Can you say if you agree with this change. It jumped out at me as being wrong so I changed it.

Comment (naknomum) [reply app/modules/sightings/resources.py:476]: i do not think we have a "standard" way to treat `op=add` vs `op=replace` when the property/path is a **list**.  so i am fine with this following the same equivalence pattern as single-value properties.   unless @brmscheiner needs a specific call to replace the old list with a new list.

Comment (naknomum) [reply app/modules/sightings/resources.py:476]: oh i see yes i think you are right.  this building of `anns` _should_ be done for either `op`.   good call.

**Pull request merged**